### PR TITLE
fixed "no private key available" issue. Part 2. add email to existing key

### PR DESCRIPTION
**Pull request by DenBond7**
This PR added the second option `add email to existing key`

issue #597
----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

**Comment by DenBond7** [comment]
For this scenario need to add the following test

1. Updating a key with a locally saved passphrase
2. Updating a key with a passphrase in RAM

**Comment by DenBond7** [comment]
How to test(only one key exists):

1. Setup the app for example for `foo@flowcrypt.test`
2. Go to `Settings -> Keys`
3. Import a new key for example for `test@flowcrypt.test`
4. Delete the key for `foo@flowcrypt.test`
5. Go to the compose screen and try to send an encrypted message 

How to test(multiple keys exist):

1. Setup the app for example for `foo@flowcrypt.test`
2. Go to `Settings -> Keys`
3. Import a new key for example for `first@flowcrypt.test`
4. Import a new key for example for `second@flowcrypt.test`
5. Delete the key for `foo@flowcrypt.test`
6. Go to the compose screen and try to send an encrypted message 

Also, the behavior will be different for case `passphraseType = KeyEntity.PassphraseType.RAM` when a private key has no passphrase in RAM.

File: FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt [hunk 1]
```diff
@@ -1856,24 +1934,49 @@ class CreateMessageFragment : BaseFragment<FragmentCreateMessageBinding>(),
...
-        DialogItem(
-          iconResourceId = R.drawable.ic_import_user_public_key,
-          title = getString(R.string.import_private_key),
-          id = RESULT_CODE_IMPORT_PRIVATE_KEY
-        ),
-        DialogItem(
-          iconResourceId = R.drawable.ic_edit_key_add_user_id,
-          title = getString(R.string.add_email_to_existing_key),
-          id = RESULT_CODE_ADD_USER_ID_TO_EXISTING_PRIVATE_KEY
+    if (account?.clientConfiguration?.usesKeyManager() == true) {
```
Comment (DenBond7) [inline FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt:1937]: @sosnovsky Please pay attention here. I've disabled the ability to use new changes for users where `account?.clientConfiguration?.usesKeyManager() == true`. I propose to discuss it a bit later and make changes via a separate PR.

**Review by sosnovsky — APPROVED** [review]
Works well 👍

File: FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt (hunk 1 above)
Comment (sosnovsky) [reply FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt:1937]: I think for now we can leave your implementation - as it's simpler, and if there will be any feedback asking for ability to fix keys even when using key manager - we'll work on it.

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenNoKeyAvailableSingleKeyWithPassphraseInDatabaseFlowTest.kt [hunk 2]
```diff
@@ -116,28 +109,10 @@ class ComposeScreenNoKeyAvailableFlowTest : BaseComposeScreenTest() {
...
-          }
-
-          return MockResponse().setResponseCode(HttpURLConnection.HTTP_NOT_FOUND)
-        }
-      })
+  @Test
+  fun testAddEmailToExistingSingleKeyPassphraseInDatabase() {
+    doTestAddEmailToExistingKey {
+      //no more additional actions
+    }
```
[the remark this answers is not in the document]
Comment (DenBond7) [reply FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenNoKeyAvailableSingleKeyWithPassphraseInDatabaseFlowTest.kt:116]: yup, it tests the simplest version(a single key with passphrase in the database)

**Pull request merged**